src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -2418,6 +2446,53 @@ unsafe object IWinRTNullableTypeDetails.GetNullableValue(IInspectable inspectabl
                 Marshal.Release(nullablePtr);
             }
         }
+
+        // Delegates are handled separately.
+        Type IWinRTNullableTypeDetails.GetNullableType() => throw new NotImplementedException();
+    }
+
+    public sealed class EnumTypeDetails<T> : IWinRTExposedTypeDetails, IWinRTNullableTypeDetails where T : struct, Enum
+    {
+        [SkipLocalsInit]
+        public ComWrappers.ComInterfaceEntry[] GetExposedInterfaces()
+        {
+            Span<ComWrappers.ComInterfaceEntry> entries = stackalloc ComWrappers.ComInterfaceEntry[2];
+            int count = 0;
+
+            if (FeatureSwitches.EnableIReferenceSupport)
+            {
+                entries[count++] = new ComWrappers.ComInterfaceEntry
+                {
+                    IID = global::WinRT.Interop.IID.IID_IPropertyValue,
+                    Vtable = ABI.Windows.Foundation.ManagedIPropertyValueImpl.AbiToProjectionVftablePtr
+                };
+
+                var type = typeof(T);

```

Comment:
Nit: don't store the type in a local, just use `typeof(T)` directly. It often leads to better codegen too, because some optimizations can break down due to the JIT not being able to properly follow the type across locals.